dev-python/colorlog: Version bump to v3.1.4

seifertm opened the pull request:
Bug: https://bugs.gentoo.org/664182

* Updated to EAPI 7
* Added `pypy` and `pypy3` to `PYTHON_COMPAT` both of which are tested by upstream
* Removed `RESRICT="test"`, since tests are now bundled in the PyPI tarball

Caveats:
* Added support for Python 3.7 to `PYTHON_COMPAT`, although it is not officially supported by upstream (due to a lack of activity). However, the package installs and tests pass with Python 3.7. There are also other packages using `colorlog` that support Python 3.7, e.g. [mode](https://pypi.org/project/mode/)
* Tests are installed into the package colorlog.tests

Ebuild tested with Python 2.7, 3.6, and 3.7 with `FEATURES="test"`.

sbraz commented on dev-python/colorlog/colorlog-3.1.4.ebuild:
```diff
@@ -0,0 +1,28 @@
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? ( dev-python/pytest[${PYTHON_USEDEP}] )"
+
+DOCS=( README.md )
```
You don't need to set that explicitly:
https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-94001r7

sbraz commented on dev-python/colorlog/colorlog-3.1.4.ebuild:
```diff
@@ -0,0 +1,28 @@
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6,7} pypy pypy3 )
```
Shorter and sorted: `pypy{,3} python{2_7,3_{4,5,6,7}}`

sbraz commented on dev-python/colorlog/colorlog-3.1.4.ebuild:
```diff
@@ -0,0 +1,28 @@
+IUSE="test"
+
+RDEPEND=""
+DEPEND="${RDEPEND}
```
`DEPEND=${RDEPEND}` is usually wrong for Python packages which don't link with their runtime dependencies. It's even weirder with an empty RDEPEND.

seifertm commented:
Thanks for the review.

I cleaned up the ebuild based on your comments.

seifertm commented:
Yes, tests pass on my machine. I will investigate the issue…

```
>>> Test phase: dev-python/colorlog-3.1.4
 * python2_7: running distutils-r1_run_phase python_test
============================================================================================================================= test session starts =============================================================================================================================
platform linux2 -- Python 2.7.14, pytest-3.2.2, py-1.4.34, pluggy-0.4.0 -- /usr/bin/python2.7
cachedir: .cache
rootdir: /var/tmp/portage/dev-python/colorlog-3.1.4/work/colorlog-3.1.4, inifile:
collecting ... collected 27 items

colorlog/tests/test_colorlog.py::test_colored_formatter PASSED
colorlog/tests/test_colorlog.py::test_custom_colors PASSED
colorlog/tests/test_colorlog.py::test_reset PASSED
colorlog/tests/test_colorlog.py::test_no_reset PASSED
colorlog/tests/test_colorlog.py::test_secondary_colors PASSED
colorlog/tests/test_colorlog.py::test_some_secondary_colors PASSED
colorlog/tests/test_colorlog.py::test_percent_style PASSED
colorlog/tests/test_colorlog.py::test_braces_style SKIPPED
colorlog/tests/test_colorlog.py::test_template_style SKIPPED
colorlog/tests/test_colorlog.py::test_ttycolorlog PASSED
colorlog/tests/test_colorlog.py::test_ttycolorlog_notty PASSED
colorlog/tests/test_config.py::test_build_from_file PASSED
colorlog/tests/test_config.py::test_build_from_dictionary PASSED
colorlog/tests/test_escape_codes.py::test_esc PASSED
colorlog/tests/test_escape_codes.py::test_reset PASSED
colorlog/tests/test_escape_codes.py::test_bold_color PASSED
colorlog/tests/test_escape_codes.py::test_fg_color PASSED
colorlog/tests/test_escape_codes.py::test_bg_color PASSED
colorlog/tests/test_escape_codes.py::test_rainbow PASSED
colorlog/tests/test_escape_codes.py::test_parse_colors PASSED
colorlog/tests/test_escape_codes.py::test_parse_multiple_colors PASSED
colorlog/tests/test_escape_codes.py::test_parse_invalid_colors PASSED
colorlog/tests/test_example.py::test_example PASSED
colorlog/tests/test_exports.py::test_exports PASSED
colorlog/tests/test_logging.py::test_logging_module PASSED
colorlog/tests/test_logging.py::test_colorlog_module PASSED
colorlog/tests/test_logging.py::test_colorlog_basicConfig PASSED
```

seifertm commented:
The pytest version seems to be the issue here. I can reproduce your errors with newer versions of pytest (tested with v3.6.3 and v3.4.2). If I downgrade to 3.2.2, the tests pass.

colorlog-3.1.4 [supports Python 2.6](https://github.com/borntyping/python-colorlog/tree/v3.1.4#compatibility). Support for Python 2.6 [was dropped in pytest-3.3](https://docs.pytest.org/en/latest/changelog.html#id222). Therefore, I suggest changing the test dependency to `<dev-python/pytest-3.3`.

I will file an issue in the upstream repo and adjust the ebuild.

seifertm commented:
* Added maximum version of dependency on pytest
* Removed Python 3.7 from PYTHON_COMPAT and referenced the blocking issue

sbraz commented:
Tests don't fail from the git repository, even when I checkout 3.1.4 or create a new sdist. I'm trying to find the culprit.

mgorny commented:
Thanks. Some plugin might be causing it, or even `-vv`, I guess.

seifertm commented:
Invoking it with `-p no:logging` works for me as well.
Thanks for looking into this!

* Added Python 3.7 to PYTHON_COMPAT
* Removed maximum version of dependency on pytest
* Added `-p no:logging` to pytest invocation
